components/scream/cime_config/namelist_defaults_scream.xml

```diff
@@ -0,0 +1,210 @@
+  </selectors>
+
+  <!--
+      The following blocks define the input files for the model.
+      There are some rules that must be followed.
+
+      Lingo:
+        Outer element: An XML element that is not intended to be an input paramter,
+                       it is only for grouping/namespacing the elements within it
+        Parameter element: An XML element representing an individual input parameter.
+                           This element should not have child elements. It should have element text
+                           representing the desired value. These elements can have
+                           one or more selectors.
+        Selector: An XML attribute that checks a property of the case. See the documentation
+                  above the <selectors> element above for more details.
+        Metadata: An XML attribute that represents metadata for an input parameter
+
+      Rules:
+        1) Parameters can have any number of parameter elements. A parameter element
+           with no selectors, which means it controls the default param value, must
+           come before parameter elements with selectors. If multiple selection elements
+           are matched, then the match the occurs last takes precendence.
+        2) The first parameter element encounter for a parameter will define the default metadata
+           for that parameter. All subsequent parameter elements will assume the same
+           metadata as the first unless an explicit override occurs.
+        3) For a parameter whose elements all have selectors, if none of the selectors
+           are matched, then the parameter will be omitted from the inputs.
+        4) Parameter types will be inferred. You can override the inferred type
+           via the 'type' metadata attribute. Types are:
+             logical: will be inferred from any case of true/false
+             integer: will be inferred from any sequence of 0-9
+             real   : will be inferred from any numeric expression with a decimal point or scientific notation
+             string : everything else
+        5) A parameter can be set to any value of the correct type unless
+           the 'valid_values' metadata attribute is set. If desired, this
+           metadata should be set up as a comma-separated list of values.
+           TODO support numeric checks (<, >, ...)??
+        6) Parameter metadata can be overriden in subsequent parameter elements, but
+           by default, metadata will be the same as the first parameter element. The
+           final metadata for a parameter will reflect what's in the last matching
+           parameter element. User changes to the case XML file must be compatible
+        7) You can inline case values into parameter values via this syntax:
+           <param>MY GRID IS ${ATM_GRID}</param>
+           This would result in parameter 'param' having the value 'MY GRID IS ne4ne4'
```

Comment:
This example confused me because creating a phrase is far from our typical usage. Perhaps switch to an example more related to our work? Like "<init_file> scream_init_${ATM_GRID}.nc</param}"?